src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationFailureException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LEGAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserNotNull(user);
+        checkUserAge(user);
+        checkUserLoginLength(user);
+        checkUserPasswordLength(user);
+        checkUserNotRegisteredYet(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUserNotNull(User user) {
+        if (user == null) {
+            throw new RegistrationFailureException("User registration failed due to user is null");
+        }
+    }
+
+    private void checkUserAge(User user) {
+        if (user.getAge() == null || user.getAge() < MIN_LEGAL_AGE) {
+            throw new RegistrationFailureException("User's age is less than"
+                    + MIN_LEGAL_AGE + " or not specified");
+        }
```

Comment:
I'd separate the null check from the other validation, to get the clear message in both cases, same for other methods